components/support/rs_client/src/http_client.rs

```diff
@@ -107,13 +227,37 @@ impl Client {
         let max_backoff = backoff.max(retry_after);
 
         if max_backoff > 0 {
-            self.remote_state.replace(RemoteState::Backoff {
+            *current_state = RemoteState::Backoff {
                 observed_at: Instant::now(),
                 duration: Duration::from_secs(max_backoff),
-            });
+            };
         }
         Ok(())
     }
+
+    fn lookup_attachments_base_url(base_url: Url) -> Result<Url> {
+        let req = Request::get(base_url);
+        let server_info = req.send()?.json::<RemoteSettingsServerInfo>()?;
+        match server_info.capabilities.attachments {
+            Some(capability) => Ok(Url::parse(&capability.base_url)?),
+            None => Err(ClientError::AttachmentsUnsupportedError),
+        }
+    }
+}
+
+#[derive(Deserialize)]
+struct RemoteSettingsServerInfo {
```

Comment:
```suggestion
struct ServerInfo {
```

(Since we're already in the `rs_client` crate, let's drop the `RemoteSettings` qualifier! 😊)

Reply:
Removing this from the private types, but leaving them in the public API for now, assuming that was the intent of this comment.

Reply:
Ahhh, sorry, nope, I meant literally renaming `RemoteSettingsServerInfo` to `ServerInfo` (and doing the same for the other `RemoteSettings`-prefixed types)!

Since we're already in a crate that deals with "Remote Settings things", I wasn't sure that the `RemoteSettings` prefix was adding a whole lot of value—and naming it `ServerInfo`, etc. would be in line with `Attachment` and `Record` 😅

Reply:
Yeah, I think we were on the same page! `RemoteSettingsResponse` might be worth keeping to avoid future naming conflicts in the iOS megazord since there's no real namespacing strategy available there AFAIK.

Reply:
Oop, yes, I'm sorry I forgot about `RemoteSettingsResponse`! :joy: Yep, we have to worry about naming conflicts for anything in UDL, but non-`pub` types (and `pub` types that aren't in UDL) in Rust are okay!